Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/task/YarnHistoryTask.java

```diff
@@ -40,29 +44,32 @@
 
     protected final IClusterDetail clusterDetail;
     protected final String path;
-    private final CredentialsProvider credentialsProvider =  new BasicCredentialsProvider();
-
+    private static final CredentialsProvider CREDENTIALS_PROVIDER =  new BasicCredentialsProvider();
+    private static final WebClient WEB_CLIENT = new WebClient();
     public YarnHistoryTask(@NotNull IClusterDetail clusterDetail, @NotNull String path, @NotNull FutureCallback<String> callback) {
         super(callback);
         this.clusterDetail = clusterDetail;
         this.path = path;
         try {
-            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(clusterDetail.getHttpUserName(), clusterDetail.getHttpPassword()));
+            CREDENTIALS_PROVIDER.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(clusterDetail.getHttpUserName(), clusterDetail.getHttpPassword()));
         } catch (HDIException e) {
             e.printStackTrace();
         }
     }
 
     @Override
     public String call() throws Exception {
-        CloseableHttpClient httpclient = HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider).build();
-        HttpGet httpGet = new HttpGet(path);
-        httpGet.addHeader("Content-Type", "text/html");
-
-        CloseableHttpResponse response = httpclient.execute(httpGet);
+        WEB_CLIENT.setCredentialsProvider(CREDENTIALS_PROVIDER);
+        HtmlPage htmlPage = WEB_CLIENT.getPage(path);
 
-        HttpEntity httpEntity = response.getEntity();
+        // parse pre tag from html response
+        // there's only one 'pre' in response
+        DomNodeList<DomElement> preTagElements = htmlPage.getElementsByTagName("pre");
 
-        return IOUtils.toString(httpEntity.getContent());
+        if (preTagElements.size() == 0) {
+            return "No logs here or logs not available";
```

Comment:
The method already throws exceptions, it might be better to construct an error html web page at the outside catching function.